batch2/create-batch-tables.sql

```diff
@@ -94,6 +118,42 @@ CREATE INDEX batch_attributes_key_value ON `batch_attributes` (`key`, `value`(25
 
 DELIMITER $$
 
+CREATE TRIGGER attempts_before_update BEFORE UPDATE ON attempts
+FOR EACH ROW
+BEGIN
+  IF OLD.end_time IS NOT NULL AND (NEW.end_time IS NULL OR NEW.end_time > OLD.end_time) THEN
```

Comment:
end_time can't be null and not null.

Reply:
I guess that's true for the end time, but I think we do need this check for the start time.

Reply:
Wait, one is OLD and the other is NEW. Is that your objection?